web/src/components/atoms/Input/index.tsx

```diff
@@ -1,6 +1,33 @@
-import { Input, InputProps } from "antd";
+import { Input as AntInput, InputProps } from "antd";
+import { useMemo } from "react";
+import { runes } from "runes2";
 
 export type { SearchProps } from "antd/lib/input";
 
+type Props = {
+  value?: string;
+} & InputProps;
+
+const Input: React.FC<Props> = ({ value, maxLength, ...props }, ref) => {
```

Comment:
I think it is better to use `forwardRef` here, you can import `InputRef` from `antd`.
Please do the same for all the other components that pass the ref from a parent component to a child component.
```suggestion
const Input = forwardRef<InputRef, Props>(({ value, maxLength, ...props }, ref) => {

```